app/Jobs/ApplicationDeploymentJob.php

```
@@ -2546,16 +2924,33 @@ private function run_post_deployment_command()
         throw new RuntimeException('Post-deployment command: Could not find a valid container. Is the container name correct?');
     }
 
+    /**
+     * Check if the deployment was cancelled and abort if it was
+     */
+    private function checkForCancellation(): void
+    {
+        $this->application_deployment_queue->refresh();
+        if ($this->application_deployment_queue->status === ApplicationDeploymentStatus::CANCELLED_BY_USER->value) {
+            $this->application_deployment_queue->addLogEntry('Deployment cancelled by user, stopping execution.');
+            throw new \RuntimeException('Deployment cancelled by user', 69420);
```

PHPMD (2.15.0)

Missing class import via use statement (line '2935', column '23'). (undefined)

(MissingImport)